BLUESPAWN-win-client/src/util/permissions/permissions.cpp

```diff
@@ -1,85 +1,87 @@
 #include "util/permissions/permissions.h"
 #include "util/log/Log.h"
+#include <lm.h>
 
-namespace Permissions{
 
-	bool AccessIncludesAll(const ACCESS_MASK& access){
+namespace Permissions {
+
+	bool AccessIncludesAll(const ACCESS_MASK& access) {
 		return ((access & GENERIC_ALL) == GENERIC_ALL) ||
 			((access & FILE_ALL_ACCESS) == FILE_ALL_ACCESS);
 	}
 
-	bool AccessIncludesWrite(const ACCESS_MASK& access){
-		return AccessIncludesAll(access) ||
+	bool AccessIncludesWrite(const ACCESS_MASK& access) {
+		return AccessIncludesAll(access) || 
 			((access & GENERIC_WRITE) == GENERIC_WRITE) ||
 			((access & FILE_GENERIC_WRITE) == FILE_GENERIC_WRITE);
```

Comment:
What about specific access rights? Not worth considering here?

Reply:
I don't think so, if dealing with specific rights that should be dealt with in the wrappers of the specific object. I.e., registry should handle any registry specific rights. That way this doesn't need to worry about the type of the object.

Reply:
In an old program of mine, I worked almost exclusively with specific access writes. For example FILE_READ grants read access to a file, but is not generic_read/generic_write. This is fine for now; most code won't use FILE_READ. I do have some code lying around that can deduce handle types we may want to integrate later